### PR TITLE
Modify SplitError's behavior.

### PR DESCRIPTION
**Pull request by pattyshack**
1. Instead of returning the unwrapped coded error, which loses a lot of error message information, always return the original error wrapped with the important error code.
2. For non-failure, use the root cause error code instead of the shallowest error code.

### CODE REVIEW

File: fvm/errors/errors.go [hunk 1]
```diff
@@ -53,19 +50,17 @@ func findImportantCodedError(err error) (CodedError, bool) {
...
+			return coded, false
 		}
 
-		if coded.Code().IsFailure() {
+		var nextCoded CodedError
+		if !As(coded.Unwrap(), &nextCoded) {
 			return coded, false
 		}
+
+		coded = nextCoded
```
Comment (zhangchiqing) [inline fvm/errors/errors.go:63]: how does the value of nextCoded get updated? shouldn't this be `coded = coded.Unwrap()`?

Comment (pattyshack) [reply fvm/errors/errors.go:63]: nextCoded is updated by `As(coded.Unwrap(), &nextCoded)`.  The error is already unwrapped to the next nested coded error

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**